src/javaharness/java/arcs/webimpl/PortablePromiseJsImpl.java

```diff
@@ -0,0 +1,38 @@
+package arcs.webimpl;
+
+import arcs.api.PortablePromise;
+import elemental2.promise.Promise;
+import elemental2.promise.Promise.PromiseExecutorCallbackFn;
+import elemental2.promise.IThenable.ThenOnFulfilledCallbackFn;
+import elemental2.promise.IThenable;
+import java.util.function.Consumer;
+
+
+class PortablePromiseJsImpl<T> implements PortablePromise<T> {
+  private final Promise<T> promise;
+
+  public PortablePromiseJsImpl(Consumer<T> resolve, Runnable reject) {
+    PromiseExecutorCallbackFn executor = new PromiseExecutorCallbackFn<T>() {
+      @Override
+      public void onInvoke(ResolveCallbackFn<T> resolve, RejectCallbackFn reject) {
+        // TODO: what should be here?
```

Comment:
I think PortablePromiseJsImpl should take a PortablePromiseExecutor interface as described before, and store it on a field. Then instead onInvoke() here you should delegate, e.g. 

portablePromiseExecutor.onInvoke(/* delegate resolver */ (T arg) -> {
   resolve.onInvoke(arg);
},
  /* delegate reject */ 
  () -> { reject.run(); }
);

That is, we pass in our own consumer/runnable to the PortablePromiseExecutor,  and when it calls conumer.accept() we invoke ResolveCallbackFn.onInvoke(value) which is the system (browser provided) resolve() function which will resolve the underlying native promise.